### ISSUE TITLE
Grid: Node not registering successfully to the hub in v3.3

### ISSUE THREAD

**Issue by oscar-qlik** [SeleniumHQ/selenium#3631]
## Meta -
OS:  Windows 7 (node), Ubuntu Server 16.04
Selenium Version:  3.3
Browser:  -
Browser Version: -

## Expected Behavior -
The node successfully registers and stays registered to the hub.
It works in 3.2.

## Actual Behavior -
The node never seems to be successfully registered at all. According to the log it is registered and then unregistered after just a few seconds. This repeats every time the node tries to register (every 5 minutes in our case).

Have a look at the logs.

### Hub
    14:15:23.737 INFO - Registered a node http://1.1.1.1:5555
    14:15:28.990 INFO - Marking the node http://1.1.1.1:5555 as down: cannot reach the node for 1 tries
    14:15:39.022 INFO - Unregistering the node http://1.1.1.1:5555 because it's been down for 10031 milliseconds
    14:15:39.022 WARN - Cleaning up stale test sessions on the unregistered node http://1.1.1.1:5555

### Node
    14:15:22.642 INFO - Launching a Selenium Grid node
    14:15:23.315 INFO - Driver class not found: com.opera.core.systems.OperaDriver
    14:15:23.315 INFO - Driver provider com.opera.core.systems.OperaDriver registration is skipped:
    Unable to create new instances on this machine.
    14:15:23.315 INFO - Driver class not found: com.opera.core.systems.OperaDriver
    14:15:23.315 INFO - Driver provider com.opera.core.systems.OperaDriver is not registered
    14:15:23.331 INFO - Driver provider org.openqa.selenium.safari.SafariDriver registration is skipped:
    registration capabilities Capabilities [{browserName=safari, version=, platform=MAC}] does not match the current platform VISTA
    14:15:23.549 INFO - Selenium Grid node is up and ready to register to the hub
    14:15:23.549 INFO - Starting auto registration thread. Will try to register every 300000 ms.
    14:15:23.549 INFO - Registering the node to the hub: http://2.2.2.2:4444/grid/register
    14:15:23.580 INFO - The node is registered to the hub and ready to use
    14:15:28.804 INFO - SessionCleaner initialized with insideBrowserTimeout 60000 and clientGoneTimeout 30000 polling every 3000
    14:20:23.845 INFO - Registering the node to the hub: http://2.2.2.2:4444/grid/register
    14:20:23.861 INFO - The node is registered to the hub and ready to use
    14:25:23.958 INFO - Registering the node to the hub: http://2.2.2.2:4444/grid/register
    14:25:23.974 INFO - The node is registered to the hub and ready to use

## Steps to reproduce -
It stopped working after upgrading from 3.2 to 3.3.

**Comment by szaluk** [comment]
I am seeing the same behavior.  I am also seeing this in the grid console:

DefaultRemoteProxy unknown version,server response code : 500

![error](https://cloud.githubusercontent.com/assets/2184567/23765623/c85a625e-04cf-11e7-892d-114dc86faa92.png)

**Comment by mach6** [comment]
This is happening because of #3618.

The hub pings back the node via the `/status` endpoint -- which is failing. So the hub marks the node down and eventually the node tries to re-register.

**Comment by szaluk** [comment]
This is fixed in 3.3.1

**Comment by krmahadevan** [comment]
@oscar-qlik - Can you please help vet out if your issue is resolved and if its working fine, please help close this issue ?

**Comment by oscar-qlik** [comment]
It works perfectly fine in 3.3.1, the nodes connect to the hub. You can close the issue.
Thank you for your support!

**Comment by michelleXIE2014** [comment]
where can we find 3.3.1? the seleniumHQ selenium version is 3.3.0.

**Comment by diemol** [comment]
@michelleXIE2014 
Just answered that [here](https://github.com/SeleniumHQ/selenium/issues/3630#issuecomment-286155046)

**Comment by michelleXIE2014** [comment]
@diemol thank you :)

**Comment by aapande** [comment]
I am still facing this issue with selenium  version 3.3.1. 
http://stackoverflow.com/questions/43394478/selenium-nodes-are-getting-reconnected-to-hub
Can someone take look at this ??

**Comment by rdbaron** [comment]
I'm still seeing this in 3.3.1 as well:

![image](https://cloud.githubusercontent.com/assets/6018498/25637289/749bb3ae-2f52-11e7-8aa3-653114b490be.png)

**Comment by szaluk** [comment]
This has been working fine for me since 3.3.1.  Just a quick thought - Can you also verify the nodes are using 3.3.1+ as well and not just the hub?  From your screen shot you can't tell what version is running on the nodes.  It should look something like this when it's connected:

![hub](https://cloud.githubusercontent.com/assets/2184567/25637507/365c5ab6-2f53-11e7-8161-00664c8a6633.png)

Thanks,
Steve

**Comment by rdbaron** [comment]
My issue is probably unrelated.  I am running my grid using Docker.  Turns out if I start some containers locally, I can't reproduce this issue.  It's only happening when I try to deploy to a Swarm.  Same symptoms though, nodes constantly re-registering and a `500` when hitting `/status`

**Comment by memoryvandel** [comment]
Hi there, I've encountered this issue on grid v3.4.0 - same prompt output as the original post.

It does register and appears to disconnect/reconnect every few minutes. In the Grid Console it reads " DefaultRemoteProxy unknown version, Connect to X:5555 [/X] failed: connect timed out " (it does state the ID and OS). Both HUB and nodes have the same grid version and I have tried with alternative versions on both machines, testing on the same machine allows scripts to be executed successfully.

Have I missed out a step in my code:

[Hub]
java -jar selenium-server-standalone-3.4.0.jar -role hub

[Node]
java -Dwebdriver.chrome.driver=C:\seleniumGrid\chromedriver.exe  -jar selenium-server-standalone-3.4.0.jar -role webdriver -hub http://<ipaddress>:4444/grid/register -port 5555  -timeout 60000 -browser browserName=chrome,maxInstances=1

any feedback would be much appreciated

**Comment by memoryvandel** [comment]
nvm, realized that I didn't replace -webdriver with -node & that my firewall prevented it launching.

**Comment by sandeepdharembra** [comment]
I am getting this issue with Selenium 3.4. Here is the complete scenario -

1)  Hub is running on my ubuntu machine
2) Ran node on the same machine as well as another machine and they both are working fine
3) Started one Win10 VM on the Ubuntu machine (running hub) using Oracle VirtualBox. Disabled Windows Firewall
4) Using the same selenium version (3.4), trying to start a node on this VM -
**Node logs say** - Node is registered to the hub and ready to  use

**Command used** : java -Dwebdriver.edge.driver="driver_path" -jar selenium-server-standalone-3.4.0.jar -role node -hub http://<ip>:4444/grid/register

**Hub Logs** -
```
19:30:14.787 INFO - Unregistering the node http://10.0.2.15:5555 because it's been down for 60030 milliseconds
19:30:14.787 WARN - Cleaning up stale test sessions on the unregistered node http://10.0.2.15:5555
19:30:15.278 INFO - Registered a node http://10.0.2.15:5555
19:30:35.290 INFO - Marking the node http://10.0.2.15:5555 as down: cannot reach the node for 2 tries
```

**Comment by prasanta-biswas** [comment]
This issues occurred in v3.5.0. Please fix it.

**Comment by barancev** [comment]
@prasanta-biswas Raise a new issue and provide logs to diagnose it.

**Comment by ManjushaQA** [comment]
Hi I have encountered this issues with selenium-server-standalone-3.8.1. Can anybody help me out..
![grid](https://user-images.githubusercontent.com/32263961/33753321-aa027f70-dc0c-11e7-8751-896e6399237d.png)

**Comment by ajinkyaapande** [comment]
I faced this issue many times. and i found out solution for the same. 
You need to allow java application through firewall on slave machine. 
Following in screenshot of the same
![image](https://user-images.githubusercontent.com/30824243/33754216-31bdf080-dc11-11e7-9fc6-3ef93e7cd775.png)

**Comment by athota13** [comment]
Hi I am encountering this issues"Marking the node http://10.3.12.86:5555 as down: cannot reach the node for 2 tries" with selenium-server-standalone-3.8.1 on MAC(os x el capitan 10.11.6). Can anybody help me out please .

**Comment by arrustamyan** [comment]
hi @athota13 . If you are running the hub and the node on the same network than you can make the hub reference the node by a given host name instead of an IP address. So
```
# assuming that there is an overlay network named *selenium* to which the hub is attached to
docker service create \
  --name chrome \
  -e HUB_PORT_4444_TCP_ADDR=selenium-hub \
  -e HUB_PORT_4444_TCP_PORT=4444 \
  -e REMOTE_HOST="http://chrome:5555" \
  --network selenium \
  selenium/node-chrome
```
After registering hub will connect to the node using `http://chrome:5555` instead of the IP which is apparently one that is not reachable.

I am using docker swarm but you can easily convert this to `docker run` syntax.

**Comment by krmahadevan** [comment]
This issue is closed. So you might want to include the following and open a
new issue.

1. Selenium version ( pls try using latest version. As of today it is 3.13)
2. Command lines being used to start node and hub.
3. Console logs from both hub and node.





-- 

Thanks & Regards
Krishnan Mahadevan

"All the desirable things in life are either illegal, expensive, fattening
or in love with someone else!"
My Scribblings @ http://wakened-cognition.blogspot.com/

My Technical Scribbings @ http://rationaleemotions.wordpress.com/

**Comment by Vinodkomeershetty** [comment]
Issue is happening with 3.8 standalone selenium version as well , can any one please take a look

**Comment by krmahadevan** [comment]
The latest released version of Selenium Grid is 3.14.1. Please retry using
that.

>
